kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/LogReplaySuite.scala

```diff
@@ -13,84 +13,264 @@
+    checkTable(
+      path = goldenTablePath("only-checkpoint-files"),
+      expectedAnswer = expectedValues.map(TestRow(_))
+    )
+  }
+
+  Seq("protocol", "metadata").foreach { action =>
+    test(s"missing $action should fail") {
+      val path = goldenTablePath(s"deltalog-state-reconstruction-without-$action")
+      val e = intercept[IllegalStateException] {
+        latestSnapshot(path).getSchema(defaultTableClient)
       }
-    } finally scanFilesBatchIter.close()
-    dataBatches
+      assert(e.getMessage.contains(s"No $action found"))
+    }
   }
 
-  test("simple end to end with inserts and deletes and checkpoint") {
-    val unresolvedPath = GoldenTableUtils.goldenTablePath("basic-with-inserts-deletes-checkpoint")
-    val table = io.delta.kernel.Table.forPath(tableClient, unresolvedPath)
-    val conf = new Configuration()
-    val client = DefaultTableClient.create(conf)
-    val snapshot = table.getLatestSnapshot(client)
-    val schema = snapshot.getSchema(client)
-    val version = snapshot.getVersion(client)
-
-    assert(schema === new StructType().add("id", LongType.LONG))
-    assert(version === 13)
-
-    var output = new scala.collection.mutable.ArrayBuffer[Long]()
-    readSnapshot(schema, snapshot).foreach { data =>
-      val rowsIter = data.getRows
-      while (rowsIter.hasNext) {
-        val row = rowsIter.next()
-        output += row.getLong(0)
+  // TODO missing protocol should fail when missing from checkpoint
+  //   GoldenTable("deltalog-state-reconstruction-from-checkpoint-missing-protocol")
+  //   generation is broken and cannot be regenerated with a non-null schemaString until fixed
+  Seq("metadata" /* , "protocol" */).foreach { action =>
+    test(s"missing $action should fail missing from checkpoint") {
+      val path = goldenTablePath(s"deltalog-state-reconstruction-from-checkpoint-missing-$action")
+      val e = intercept[IllegalStateException] {
+        latestSnapshot(path).getSchema(defaultTableClient)
       }
-      rowsIter.close()
+      assert(e.getMessage.contains(s"No $action found"))
     }
+  }
+
+  test("fetches the latest protocol and metadata") {
+    val path = goldenTablePath("log-replay-latest-metadata-protocol")
+    val snapshot = latestSnapshot(path)
+    val scanStateRow = snapshot.getScanBuilder(defaultTableClient).build()
+      .getScanState(defaultTableClient)
+
+    // schema is updated
+    assert(ScanStateRow.getLogicalSchema(defaultTableClient, scanStateRow)
+      .fieldNames().asScala.toSet == Set("col1", "col2")
+    )
 
-    output = output.sorted
-    println(output)
-    assert(output.length == 41)
+    // check protocol version is upgraded
+    val readerVersionOrd = scanStateRow.getSchema().indexOf("minReaderVersion")
+    val writerVersionOrd = scanStateRow.getSchema().indexOf("minWriterVersion")
+    assert(scanStateRow.getInt(readerVersionOrd) == 3 && scanStateRow.getInt(writerVersionOrd) == 7)
   }
 
+  test("standalone DeltaLogSuite: 'checkpoint'") {
```

Comment:
nit: why the name `standalone DeltaLogSuite`?

Reply:
I guess just noting the origin